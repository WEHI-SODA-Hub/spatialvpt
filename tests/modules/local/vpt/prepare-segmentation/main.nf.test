nextflow_process {

    name "Test Process PREPARE_SEGMENTATION"
    script "modules/local/vpt/prepare-segmentation/main.nf"
    process "PREPARE_SEGMENTATION"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/data/example_analysis_algorithm/cellpose_default_1_ZLevel.json', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images/micron_to_mosaic_pixel_transform.csv', checkIfExists: true) ]
                ])
                input[1] = 2400
                input[2] = 200
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
