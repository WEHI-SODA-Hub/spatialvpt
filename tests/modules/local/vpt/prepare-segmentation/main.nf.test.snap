{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "segmentation_specification.json:md5,cecc5abf184cf6434034650735afea07"
                    ]
                ],
                "1": [
                    "versions.yml:md5,da001a9b9e4e5a439bea2fe6ed119ec8"
                ],
                "specification_json": [
                    [
                        {
                            "id": "small_set"
                        },
                        "segmentation_specification.json:md5,cecc5abf184cf6434034650735afea07"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,da001a9b9e4e5a439bea2fe6ed119ec8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-25T11:11:53.210220656"
    }
}